### ISSUE TITLE
Mouse click ... in Flycheck errors mode reports button-activate: Symbol's function definition is void: nil

### ISSUE THREAD

**Issue by xuchunyang** [flycheck/flycheck#1814]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description

Mouse click "..." in `*Flycheck errors*` reports 

> button-activate: Symbol's function definition is void: nil 

![Screen Shot 2020-08-01 at 00 14 13](https://user-images.githubusercontent.com/4550353/89054772-f4407000-d38b-11ea-82e1-687e204cebbb.png)

Mouse click the "no-" part is ok, it opens https://eslint.org/docs/rules/no-unused-vars. and keyboard RET key works totally fine.

## Steps to reproduce

1. Use eslint to check this file with flycheck

```js
const hello = () => {
  const x = 42;
  console.log('Hello, World!')
}

hello()
```

2. M-x flycheck-list-errors to open `*Flycheck errors*` then mouse click "..."

here is the backtrace after M-x toggle-debug-on-error

> Debugger entered--Lisp error: (void-function nil)
>   (nil ("..." . 2))
>   (button-activate ("..." . 2) t)
>   (push-button (mouse-2 (#<window 96 on *Flycheck errors*> 33 (260 . 7) 4905903 ("..." . 2) 33 (28 . 0) nil (3 . 7) (9 . 19))))
>   (funcall-interactively push-button (mouse-2 (#<window 96 on *Flycheck errors*> 33 (260 . 7) 4905903 ("..." . 2) 33 (28 . 0) nil (3 . 7) (9 . 19))))
>   (call-interactively push-button nil nil)
>   (command-execute push-button)

- [ ] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## System configuration
```
Syntax checkers for buffer hello.js in js-mode:

First checker to run:

  javascript-eslint
    - may enable:  yes
    - executable:  Found at /Users/xcy/bin/npx-eslint
    - config file: found

Checkers that are compatible with this mode, but will not run until properly configured:

  javascript-jshint (automatically disabled) reset
    - may enable:         no
    - executable:         Not found
    - configuration file: Not found

  javascript-standard (automatically disabled) reset
    - may enable: no
    - executable: Not found

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20200610.1809)
Emacs version:    27.1
System:           x86_64-apple-darwin19.5.0
Window system:    mac
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

This is a small issue, I found it by accidient, the priority should be very low.

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by cpitclaudel** [comment]
Can you try on Emacs master?   think this might be the same as https://debbugs.gnu.org/cgi/bugreport.cgi?bug=40859

**Comment by xuchunyang** [comment]
I didn't try on Emacs master but I do apply [your patch](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=40859;msg=17;att=1;filename=0001-Only-treat-display-strings-as-buttons-if-they-have-a.patch) manually to the `push-button` function, then the problem is gone.

**Comment by cpitclaudel** [comment]
Great, thanks a lot for confirming.  OK to close this then?
